Author: Jordi Mallach <jordi@debian.org>
Description: Check for GNU.
 Fix builds on hurd-i386 by checking for __GNU__ and adding a gnu TARGETOS
 that defines NO_AFFINITY_NP.
Forwarded: no

Index: mame/makefile
===================================================================
--- mame.orig/makefile	2013-02-09 17:32:21.000000000 +0100
+++ mame/makefile	2013-02-13 22:40:30.000000000 +0100
@@ -77,6 +77,9 @@
 ifeq ($(firstword $(filter Linux,$(UNAME))),Linux)
 TARGETOS = linux
 endif
+ifeq ($(firstword $(filter GNU,$(UNAME))),GNU)
+TARGETOS = gnu
+endif
 ifeq ($(firstword $(filter Solaris,$(UNAME))),Solaris)
 TARGETOS = solaris
 endif
Index: mame/src/osd/sdl/sdl.mak
===================================================================
--- mame.orig/src/osd/sdl/sdl.mak	2013-01-28 09:49:47.000000000 +0100
+++ mame/src/osd/sdl/sdl.mak	2013-02-13 22:42:27.000000000 +0100
@@ -284,6 +284,13 @@
 
 endif
 
+ifeq ($(TARGETOS),gnu)
+BASE_TARGETOS = unix
+SYNC_IMPLEMENTATION = tc
+SDL_NETWORK = taptun
+DEFS += -DNO_AFFINITY_NP
+endif
+
 ifeq ($(TARGETOS),win32)
 BASE_TARGETOS = win32
 SYNC_IMPLEMENTATION = win32
Index: mame/src/osd/sdl/sdlprefix.h
===================================================================
--- mame.orig/src/osd/sdl/sdlprefix.h	2013-01-11 08:32:46.000000000 +0100
+++ mame/src/osd/sdl/sdlprefix.h	2013-02-13 22:43:27.000000000 +0100
@@ -42,7 +42,7 @@
 /* Large file support on IRIX needs _SGI_SOURCE */
 #undef _POSIX_SOURCE
 
-#elif defined(__linux__) || defined(__FreeBSD_kernel__)
+#elif defined(__linux__) || defined(__GNU__) || defined(__FreeBSD_kernel__)
 #define SDLMAME_LINUX 1
 
 #elif defined(__FreeBSD__)
